json_schema_fuzz/schema_operations.py

```diff
@@ -1,45 +1,245 @@
+
+            # Merge and save
+            denested_schemas = [
+                subschema for outer_list in current_permutation
+                for subschema in outer_list
+            ]
+            new_anyof_values.append(merge(*denested_schemas))
+
+        # If there are already anyof values,
+        # call merge() again to merge them with the
+        # new anyOf values
+        if 'anyOf' in merged_schema:
+            merged_schema["anyOf"] = merge(
+                {"anyOf": merged_schema["anyOf"]},
+                {"anyOf": new_anyof_values}
+            )
         else:
-            raise NotImplementedError(
-                f"Conflicting key {key} encountered in path {path}")
-    if in_place:
-        return None
-    else:
-        return schema_a
+            merged_schema["anyOf"] = new_anyof_values
+
+    # Numbers
+
+    minimum_values = get_val_or_none(schemas, "minimum")
+    if minimum_values:
+        merged_schema["minimum"] = max(minimum_values)
+    exclusive_minimum_values = get_val_or_none(schemas, "exclusiveMinimum")
+    if exclusive_minimum_values:
+        merged_schema["exclusiveMinimum"] = max(exclusive_minimum_values)
+
+    maximum_values = get_val_or_none(schemas, "maximum")
+    if maximum_values:
+        merged_schema["maximum"] = min(maximum_values)
+    exclusive_maximum_values = get_val_or_none(schemas, "exclusiveMaximum")
+    if exclusive_maximum_values:
+        merged_schema["exclusiveMaximum"] = min(exclusive_maximum_values)
+
+    multiple_of_values = get_val_or_none(schemas, "multipleOf")
+    if multiple_of_values:
+        merged_schema["multipleOf"] = lcm(multiple_of_values)
+
+    not_multiple_of_values = get_val_or_none(schemas, "notMultipleOf")
+    if not_multiple_of_values:
+        merged_schema["notMultipleOf"] = []
+        # notMultipleOf values might be list or number
+        for not_multiple_of_value in not_multiple_of_values:
+            if isinstance(not_multiple_of_value, list):
+                merged_schema["notMultipleOf"].extend(not_multiple_of_value)
+            else:
+                merged_schema["notMultipleOf"].append(not_multiple_of_value)
+
+    # Strings
+
+    min_length_values = get_val_or_none(schemas, "minLength")
+    if min_length_values:
+        merged_schema["minLength"] = max(min_length_values)
+
+    max_length_values = get_val_or_none(schemas, "maxLength")
+    if max_length_values:
+        merged_schema["maxLength"] = min(max_length_values)
+
+    # Object
+
+    properties_values = get_val_or_none(schemas, "properties")
+    if properties_values:
+        merged_schema["properties"] = {}
+        all_keys = set().union(*(d.keys() for d in properties_values))
+        for key in all_keys:
+            all_values = [d.get(key, {}) for d in properties_values]
+            merged_schema["properties"][key] = merge(*all_values)
+
+    required_values = get_val_or_none(schemas, "required")
+    if required_values:
+        merged_schema["required"] = []
```

Comment:
Here we definitely want
```python
merged_schema["required"] = [
    required_property
    for required_value in required_values
    for required_property in required_value
]
```
We could also make a generic `flatten_list` function if that would read better.